test-lit/lit_port.py

```diff
@@ -0,0 +1,256 @@
+""" lit_port.py
+
+    Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+    See https://llvm.org/LICENSE.txt for license information.
+    SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+"""
+
+import argparse
+import os
+import re
+import shlex
+import sys
+import subprocess
+from pathlib import Path
+
+from test_splitter import TestSplitter
+
+ROOT = Path.cwd()
+while ROOT.name != "f18":
+    ROOT = ROOT.parent
+DEFAULT_TEMPLATE = "! RUN: %flang {} %s 2>&1\n"
+ERROR_TEMPLATE = "! RUN: not %flang {} %s 2>&1 | FileCheck --check-prefixes={} %s\n"
```

Comment:
Is the redirection because the driver does not support "-o -" ?

Reply:
I'm believe `-o` is supported, but isn't needed for these types of tests. Probably will be needed for semantics modfile tests.

Reply:
Sorry, my original comment is wrong since these are error tests and errors go to the error stream and needs to be redirected to the output stream before piping into filecheck so that the error message can be compared.